CMakeLists.txt

```diff
@@ -429,6 +423,43 @@ else()
   set(PLUGIN_LLVM_VERSION_MAJOR 0)
 endif()
 
+set(DEFAULT_GPU_ARCH "" CACHE STRING "(Deprecated, use DEFAULT_TARGETS instead) Optional: Default GPU architecture to compile for when targeting GPUs (e.g.: sm_60 or gfx900)")
+set(DEFAULT_TARGETS "" CACHE STRING "Default targets to compile for")
+
+if(NOT DEFAULT_TARGETS)
+  if(DEFAULT_PLATFORM)  
+    message(DEPRECATION "DEFAULT_PLATFORM is deprecated; use DEFAULT_TARGETS instead.")
+    
+    if(DEFAULT_PLATFORM STREQUAL "cpu")
+      set(DEFAULT_TARGETS "omp")
+    endif()
+    
+    if(DEFAULT_GPU_ARCH)
+      message(DEPRECATION "DEFAULT_GPU_ARCH is deprecated; use DEFAULT_TARGETS instead.")
+
+      if(DEFAULT_PLATFORM STREQUAL "cuda")
+        set(DEFAULT_TARGETS "cuda:${DEFAULT_GPU_ARCH}")
+      elseif(DEFAULT_PLATFORM STREQUAL "rocm")
+        set(DEFAULT_TARGETS "hip:${DEFAULT_GPU_ARCH}")
+      endif()
+      
+    endif()
+  elseif(DEFAULT_GPU_ARCH)
+    message(DEPRECATION "DEFAULT_GPU_ARCH is deprecated; use DEFAULT_TARGETS instead.")
+    set(DEFAULT_TARGETS ${DEFAULT_GPU_ARCH})
```

Comment:
Shall we instead throw an error? I don't think there are any valid values for `DEFAULT_GPU_ARCH` that are also valid as valid targets (without specifying `platform:` prefix)

Reply:
```suggestion
    message(SEND_ERROR "DEFAULT_GPU_ARCH cannot be used without specifying DEFAULT_PLATFORM.")
```